### PR TITLE
Bug 1288415 - Prevent double scheduling

### PR DESCRIPTION
**Pull request by rail**

### CODE REVIEW

File: bbb/services.py [hunk 1]
```diff
@@ -513,8 +513,14 @@ def handlePending(self, data, msg):
         # except update our own runId. If we created a new BuildRequest for it
         # we'd end up with an extra Build.
         if our_task:
```
Comment (rail) [inline bbb/services.py]: As a possible scenario:
* move `our_task = ...` here
* add a write lock before querying the DB
* release the lock

File: bbb/servicebase.py [hunk 2]
```diff
@@ -18,6 +19,18 @@
 ListenerServiceEvent = namedtuple("ListenerServiceEvent", ("exchange", "routing_key", "callback", "queue_name"))
 
 
+@contextmanager
+def lock_table(db, table_name):
+
+    try:
+        if "mysql" in db.url.get_backend_name():
+            db.execute(sa.text("LOCK TABLE {} WRITE;".format(table_name)))
```
Comment (catlee) [inline bbb/servicebase.py]: Can you use SA's parameterization here? Something like 
```
db.execute(sa.text('LOCK TABLE :table WRITE;'), table_name)
```

Comment (rail) [inline bbb/servicebase.py]: I have a vague memory that you cannot use it for table names... Let me double check this.

**Comment by catlee** [comment]
The rest looks fine

File: bbb/servicebase.py (hunk 2 above)
Comment (rail) [inline bbb/servicebase.py]: Yeah, SA won't let me use `:table` for table names. I get syntax errors. Tested using
```python
db.execute("SELECT * FROM :t", "tasks").fetchall()
```

Also http://stackoverflow.com/questions/13124572/binding-variable-to-table-name-with-cx-oracle tells me that I'm not the only one.

Comment (catlee) [inline bbb/servicebase.py]: alright then! :shipit: !

**Pull request merged**